### DISCUSSION TITLE
Is there any kind of roadmap?

### DISCUSSION THREAD

**Discussion by Baael** [blakeblackshear/frigate#2338] [Ideas]
Hi there :) Is there any kind of roadmap or vision what do you want to achieve in long term? I am asking cause I am thinking about multiple things that can be helpful, right now I am doing few things in parallel and reinventing the well in some places. 
So for example what I would love to see in frigate:

- adding template matching, cameras are mostly static, that way we can use some comparator or template matching to check if for example garage doors are open, car exists in specific place, light bulb is on, child is in bed, even sun amount etc
- GUI for settings
- realtime camera filters - for example rotate (I have few cameras in portrait mode, cause my corridor is very long, adjusting it via options is painful 
- extracting detection engine and allowing to use custom models and chain it or made it parallel. I guess you did a good job already with handling events, entities, ffmpeg mqtt, etc, now some kind of unified engine blocks can be chained to allow use of yamnet, mentioned template matching etc. That way frigate may become beast and change home automation radically. 
- real time update of zones, masks, or template areas
- virtual cameras/camera scenes, right now to reuse cameras I am simply duplicating them, turning off `camera_salon_1`, changing PTZ preset, activating `camera_salon_2` with different masks and zones so I have better angle for some elements, and doing like that multiple times, would be nice to have such functionality out of the box with one trigger. 

What do you think?

**Comment by blakeblackshear** [comment]
I have a rough roadmap in my head. Most of these ideas have been thrown around before and are still things I am considering. I need to take the time to lay out a more complete roadmap to guide others interested in contributing.

**Comment by maglub** [comment]
Hi, stumbled over this one while trying to figure out if this project has a roadmap.

This is in no way negative critique, I am just curious how the project is managed, and when to expect a new minor/major release is done.

For example, the LDAP toolbox project for self-service-passwords use the milestones quite well, adding issues to a future release. They don't release a future major until all issues in the milestone is closed (or moved to another milestone), see https://github.com/ltb-project/self-service-password/milestones. I am not sure if they have documented this development process anywhere, though.

Since the documentation for this project is so good, the project management process might already be documented somewhere and I just haven't found it.

In any case, thanks for an amazing piece of software!

**Reply by NickM-27** [reply]
We have releases planned behind the scenes for what we want to accomplish, but since this is not our job it can depend on timing and what makes sense with changes that others are contributing. The general trend has been 2 major releases per year, 0.15 is getting close to first beta

**Reply by maglub** [reply]
Thanks a lot for the quick answer!

I am also involved in a handful of "free as in beer" projects, and am very much aware that it from time to time can be frustrating with unfair expectations.

When/if you ever start publishing your release plan here in Github in any shape or form is of course up to you!

Again, thanks a lot for your efforts!

**Reply by NickM-27** [reply]
Worth noting that there is a frigate+ roadmap https://github.com/orgs/frigate-nvr/projects/2/views/1

**Reply by izzygold** [reply]
oh wow, cool.

Do you think AI assistant labeling could include face detection?

**Reply by NickM-27** [reply]
Face detection is already supported in frigate+

If you are referring to face recognition that is separate and entirely unrelated to what the ai assisted labeling means

**Reply by izzygold** [reply]
I meant face recognition. Thanks for your answer
